desk/app/channels.hoon

```diff
@@ -961,6 +961,11 @@
   |=  [agent=term =path]
   ^-  ^path
   (welp /(scot %p our.bowl)/[agent]/(scot %da now.bowl) path)
+++  get-vessel
+  |=  [=flag:g =ship]
+  =/  =path
+    /groups/(scot %p p.flag)/[q.flag]/fleet/(scot %p ship)/vessel/noun
+  .^(vessel:fleet:g %gx (scry-path %groups path))
```

Comment:
This probably wants to do an existence check scry (`/exists/[p.flag]/[q.flag]`?) before doing this one. Even if you generally expect to be operating on groups that exist (because you're getting channel activity for them) you're still risking a full-on crash if that assumption ever gets broken.

Running additional `.^`s on processing every individual post will hurt performance, scry overhead remains non-trivial. You might consider wrapping a `~+` around `+get-vessel` so that we don't re-run the same scries within the same event. Alternatively, add a `/fleet` endpoint to groups agent so that we can scry and cache a single endpoint.